File: be/models/project.js
import { executar } from "../db/database.js";

class Project {
  static async getAll() {
    const [rows] = await executar("SELECT * FROM projects");
    return rows;
  }

  static async getById(id) {
    const [rows] = await executar("SELECT * FROM projects WHERE id = ?", [id]);
    return rows[0] || null;
  }

  static async create({ name, totalHours, description, cell, client, service }) {
    const result = await executar(
      "INSERT INTO projects (name, total_hours, description, cell, client, service) VALUES (?, ?, ?, ?, ?, ?)",
      [name, totalHours, description, cell, client, service]
    );
    return { 
      id: result.insertId, 
      name, 
      total_hours: totalHours, 
      description, 
      cell, 
      client, 
      service 
    };
  }

  static async delete(id) {
    await executar("DELETE FROM projects WHERE id = ?", [id]);
    return true;
  }
}

export default Project;

File: sistema-gestao-backend/src/models/project.py
from .user import db
from datetime import datetime

class Project(db.Model):
    __tablename__ = 'projects'
    
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(200), nullable=False)
    description = db.Column(db.Text)
    total_hours = db.Column(db.Integer, default=0)
    cell = db.Column(db.Integer)
    client = db.Column(db.Integer)
    service = db.Column(db.Integer)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    # Relacionamento com lançamentos de tempo
    time_entries = db.relationship('TimeEntry', backref='project', lazy=True, cascade='all, delete-orphan')
    
    def to_dict(self):
        return {
            'id': self.id,
            'name': self.name,
            'description': self.description,
            'totalHours': self.total_hours,
            'cell': self.cell,
            'client': self.client,
            'service': self.service,
            'createdAt': self.created_at.isoformat() if self.created_at else None
        }
    
    def __repr__(self):
        return f'<Project {self.name}>'

